### ISSUE TITLE
Wrong placement when an ancestor/parent container uses transform and translate

### ISSUE THREAD

**Issue by fizzylemonade** [floating-ui/floating-ui#1488]
# Background

We have a custom select component (Web component using ShadowDOM) that uses Popper.js internally for placement of a dropdown container. The custom component has a trigger element and a container/popper element that basically gets fed into `createPopper` like this:

```js
let popperInstance = createPopper(
  triggerElement,
  popperElement,
  {
    placement: "bottom-start",
    strategy: "fixed"
  });
```

Under normal circumstances this works perfect and as it should. The problem arises when this custom element has an ancestor or parent that uses a CSS transform (eg. transform: translate(x, y)). It seems like the placement logic doesn't account for transform/translate happening to parents, so that the placement is off by the sum of all of them.

Say I have a custom element named `<my-foo>` and it is placed in the DOM like this:

```html
<html>
...
  <div class="some-container">
     <my-foo></my-foo>
  </div>
...
</html>
```

Where the CSS for `some-container` looks like this:
```css
.some-container {
  position: fixed; /* or absolute */
  transform: translate(100px, 100px);
}
```

Then when the trigger element inside `my-foo` gets clicked the placement of popper element will be off by +100px in both directions (x and y).

The funny thing is that if we change the CSS above (for the container) to use left and top positions instead of transform it get placed correctly:

```css
.some-container {
  position: fixed; /* or absolute */
  left: 100px;
  top: 100px;
}
```

I've made a repro here: https://codesandbox.io/s/sharp-stitch-5woo8

Screenshot of the repro:

<img width="461" alt="placement-issue" src="https://user-images.githubusercontent.com/11876758/147949830-07fda15c-2f23-4b0d-9aa7-3611cf925a34.png">

This issue applies to version 2.11.0 of Popper, but other version my be affected as well.

**Comment by jcfranco** [comment]
Interesting. Would it be possible to work around this by using a custom modifier that adjusts positioning based on all ancestor transforms?

I haven't played much with custom modifiers, but I [prototyped a naive one](https://codesandbox.io/s/patient-glade-37z4f?file=/src/index.js) for the original use case. It is by no means production-ready and totally breaks as soon as you modify the test case. More of a proof of concept. 😅

**Comment by jcfranco** [comment]
Upon closer inspection, it seems that the placement is incorrect when there's shadow DOM involved:

w/ shadow DOM ❌
https://codepen.io/jcfranco/pen/podrpaR

w/o shadow DOM ✅
https://codepen.io/jcfranco/pen/BamdJEK

I submitted #1535 with a potential fix.

**Comment by fizzylemonade** [comment]
@jcfranco Yes, I discovered that before I registered the issue. Transforms (translate) worked fine without ShadowDOM. I could have been a bit more clearer in the problem text. Great that you found a potential fix!

**Comment by fizzylemonade** [comment]
This issue really bit me in the ass when I first discovered it. We made a generic popover control (using web component and ShadowDOM) to for example pop up some settings for a page. The problem started when we put a component inside this generic control that also used popper for placement (a custom select component using ShadowDOM in our case). Since the `computeStyle` modifier by default uses `gpuAcceleration` (transform) things got misplaced. 

Luckily, the work around was to turn off `gpuAcceleration` to revert to inset placement instead. It's not optimal, but it works.
